apis/bigqueryanalyticshub/v1alpha1/listing_types.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var BigQueryAnalyticsHubListingGVK = GroupVersion.WithKind("BigQueryAnalyticsHubDataExchangeListing")
+
+// BigQueryAnalyticsHubListingSpec defines the desired state of BigQueryAnalyticsHubDataExchangeListing
+// +kcc:proto=google.cloud.bigquery.analyticshub.v1.Listing
+type BigQueryAnalyticsHubListingSpec struct {
+	// Required. Human-readable display name of the listing. The display name must
+	//  contain only Unicode letters, numbers (0-9), underscores (_), dashes (-),
+	//  spaces ( ), ampersands (&) and can't start or end with spaces. Default
+	//  value is an empty string. Max length: 63 bytes.
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Optional. Short description of the listing. The description must not
+	//  contain Unicode non-characters and C0 and C1 control codes except tabs
+	//  (HT), new lines (LF), carriage returns (CR), and page breaks (FF). Default
```

Comment:
Nit: Reading this, I have no idea what characters are allowed 😂   Nit because I think we should probably fix that in the official proto (if at all!).